File: Darts/src/code/bean/X01ScoreRenderer.java
package code.bean;

import java.awt.Color;

import bean.AbstractTableRenderer;
import code.utils.DartsColour;

public class X01ScoreRenderer extends AbstractTableRenderer<Integer>
{
	@Override
	public Object getReplacementValue(Integer object)
	{
		return object;
	}
	
	@Override
	public void setCellColours(Integer score, boolean isSelected)
	{
		Color fg = DartsColour.getScorerForegroundColour(score);
		Color bg = DartsColour.getScorerBackgroundColour(score);
		
		if (isSelected)
		{
			fg = Color.WHITE;
			bg = DartsColour.getDarkenedColour(bg);
		}
		
		setForeground(fg);
		setBackground(bg);
	}
}

File: src/main/kotlin/dartzee/bean/X01ScoreRenderer.kt
package dartzee.bean

import dartzee.core.bean.AbstractTableRenderer
import dartzee.utils.DartsColour
import java.awt.Color

class X01ScoreRenderer : AbstractTableRenderer<Int>() {
    override fun getReplacementValue(value: Int) = value

    override fun setCellColours(typedValue: Int?, isSelected: Boolean) {
        val score = typedValue!!
        var fg = DartsColour.getScorerForegroundColour(score.toDouble())
        var bg = DartsColour.getScorerBackgroundColour(score.toDouble())

        if (isSelected) {
            fg = Color.WHITE
            bg = DartsColour.getDarkenedColour(bg)
        }

        foreground = fg
        background = bg
    }
}
